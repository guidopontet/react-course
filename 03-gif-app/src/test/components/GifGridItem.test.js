import { shallow } from "enzyme";

import GifGridItem from "../../components/GifGridItem";


describe('GifFridItem', () => {
  const title = 'Title example';
  const url = 'https://example.com/image.png';
  const wrapper = shallow( <GifGridItem title={ title } url={ url }/> );

  test('should show component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render title on a paragraph', () => {
    const p = wrapper.find('p');

    expect(p.text().trim()).toBe(title);
  });

  test('should render image with url and title', () => {
    const img = wrapper.find('img');

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });
});
