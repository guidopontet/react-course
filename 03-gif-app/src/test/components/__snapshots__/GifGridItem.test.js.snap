// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifFridItem should show component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title example"
    src="https://example.com/image.png"
  />
  <p>
     
    Title example
     
  </p>
</div>
`;
